File: Games/Dennis/src/de/hsh/dennis/model/NpcLogic/actors/Package.java
package de.hsh.dennis.model.NpcLogic.actors;

import de.hsh.dennis.model.NpcLogic.NPCEnums;

import static de.hsh.dennis.model.NpcLogic.SkinConfig.Package.skin_standard_path;

public class Package extends Npc {

    private static final int defaultSpeed = 1;

    public Package( NPCEnums.Spawn spawnType ) {
        super( skin_standard_path, spawnType, NPCEnums.NpcType.PACKAGE );
    }

    public Package( NPCEnums.Spawn spawnType, double spawnTime ) {
        super( skin_standard_path, spawnType, NPCEnums.NpcType.PACKAGE, spawnTime, defaultSpeed );
    }

    public Package( NPCEnums.Spawn spawnType, double spawnTime, double speed ) {
        super( skin_standard_path, spawnType, NPCEnums.NpcType.PACKAGE, spawnTime, speed );
    }
}

File: Games/Dennis/src/de/hsh/dennis/model/NpcLogic/actors/Package.kt
package de.hsh.dennis.model.NpcLogic.actors

import common.actor.Direction
import common.config.WindowConfig
import de.hsh.dennis.model.NpcLogic.NPCEnums.Spawn
import de.hsh.dennis.model.NpcLogic.SkinConfig

class Package(spawnType: Spawn, spawnTime: Double, speed: Double) :
        Npc(SkinConfig.Package.skin_standard_path,
            spawnType,
            spawnTime,
            speed) {

    init {
        x = if (spawnType == Spawn.RIGHT) WindowConfig.window_width.toDouble()
        else -SkinConfig.Package.skin_width.toDouble()
        y = SkinConfig.Player.posY
    }

    override fun collidesWithPlayer(player: Player): Boolean =
            player.direction === Direction.Down &&
            player.doesCollide(this)
}
